### ISSUE TITLE
Details (summary) buttons hover and redesign

### ISSUE THREAD

**Issue by dhinrichs-scottlogic** [ScottLogic/prompt-injection#783]
# Feature Request

## Description
Give the details buttons of the config hover styles. 
- Change the `details` and `summary` elements to custom buttons. Turns out semantic html isn't always better. Here's [an article](https://www.scottohara.me/blog/2022/09/12/details-summary.html) on this.
- Make the visually-hidden h2 "ScottBrew System Access" visible
- (if possible) add an "expand all/collapse all" button right next to the heading, allowing users to expand/collapse all content at once, to make the content searchable.

## Additional context
- Some screen readers read out the small triangles that come with detail and summary elements by default. You could remove them using css, but that then sometimes removes the state indication (expanded/collapsed) as well. 

## Acceptance criteria
**GIVEN** a screen reader user moves through the configurations
**WHEN** they land on the "Defence Configuration" or "details" button
**THEN** the visual icon is not announced, but the state of the button (expanded/collapsed) is announced .

**Comment by chriswilty** [comment]
@dhinrichs-scottlogic I made a comment on Anna's PR (https://github.com/ScottLogic/prompt-injection/pull/781#discussion_r1462995384) about creating a reuseable component, so you'd only need to put the styles in one place. Following that PR, we now have some duplicated styles for the Defence and Model config details elements, so perhaps you could tackle it here instead.
